File: selenium_java/LoginPrueba.java
import java.util.Map;

import org.junit.After;
import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.chrome.ChromeOptions;

import pages.LoginPagina;

public class LoginPrueba {

    private WebDriver driver;
    private LoginPagina paginaLogin;

    @Before
    public void configurar() {
        ChromeOptions opciones = new ChromeOptions();
        opciones.setExperimentalOption("prefs", Map.of(
            "credentials_enable_service", false,
            "profile.password_manager_enabled", false,
            "profile.password_manager_leak_detection", false
        ));
        opciones.addArguments("--disable-save-password-bubble");

        driver = new ChromeDriver(opciones);
        driver.manage().window().maximize();
        driver.get("https://www.saucedemo.com/");
        paginaLogin = new LoginPagina(driver);
        pausa(1000);
    }

    // probar login exitoso
    @Test
    public void loginExitoso() {
        paginaLogin.iniciarSesion("standard_user", "secret_sauce");
        paginaLogin.esperarInventario();
        pausa(2000);
        Assert.assertTrue("URL no contiene inventario", driver.getCurrentUrl().contains("inventory"));
    }

    // probar login invalido
    @Test
    public void loginFallido() {
        paginaLogin.iniciarSesion("user_invalid", "djksadjskla");
        String error = paginaLogin.obtenerMensajeError();
        pausa(2000);
        Assert.assertTrue("Mensaje de error incorrecto",
            error.contains("Username and password do not match"));
    }

    @After
    public void cerrar() {
        if (driver != null) driver.quit();
    }

    private void pausa(long ms) {
        try { Thread.sleep(ms); } catch (InterruptedException e) { Thread.currentThread().interrupt(); }
    }
}

File: LoginPrueba.py
import unittest
import time
from selenium import webdriver
from selenium.webdriver.chrome.options import Options
from paginas.LoginPagina import LoginPagina


class LoginPrueba(unittest.TestCase):

    def setUp(self):
        opciones = Options()
        opciones.add_experimental_option("prefs", {
            "credentials_enable_service": False,
            "profile.password_manager_enabled": False,
            "profile.password_manager_leak_detection": False
        })
        opciones.add_argument("--disable-save-password-bubble")

        # Para usar chromedriver.exe manual descomentar y comentar la linea de abajo
        # from selenium.webdriver.chrome.service import Service
        # self.driver = webdriver.Chrome(service=Service("../drivers/chromedriver.exe"), options=opciones)
        self.driver = webdriver.Chrome(options=opciones)
        self.driver.maximize_window()
        self.driver.get("https://www.saucedemo.com/")
        self.pagina_login = LoginPagina(self.driver)
        time.sleep(1.0)

    # probar login exitoso
    def test_login_exitoso(self):
        self.pagina_login.iniciar_sesion("standard_user", "secret_sauce")
        self.pagina_login.esperar_inventario()
        time.sleep(2.0)
        self.assertIn("inventory", self.driver.current_url, "URL no contiene inventario")

    # probar login invalido
    def test_login_fallido(self):
        self.pagina_login.iniciar_sesion("user_invalid", "djksadjskla")
        error = self.pagina_login.obtener_mensaje_error()
        time.sleep(2.0)
        self.assertIn("Username and password do not match", error, "Mensaje de error incorrecto")

    def tearDown(self):
        if self.driver:
            self.driver.quit()
